src/main/java/org/folio/dao/impl/FileDefinitionDaoImpl.java

```diff
@@ -69,4 +100,10 @@ public static Criteria constructCriteria(String jsonbField, String value) {
     criteria.setVal(value);
     return criteria;
   }
+
+  private String getFilterByUpdatedDateAndNotEmptySourcePath(Date date) {
+    String queryFilterTemplate = "WHERE LENGTH(%s.jsonb ->> 'sourcePath') > 0 AND TO_TIMESTAMP(%s.jsonb -> 'metadata' ->> 'updatedDate', 'YYYY-MM-DD HH24:MI:SS') <= '%s'";
```

Comment:
and when you remove the deprecated method for criteria this could be something like
```suggestion
      Criterion criterion = new Criterion();
      Criteria c1 = new Criteria();
      c1.addField("'sourcePath'").setOperation("IS NOT NULL");
      Criteria c2 = new Criteria();
      c2.addField("'metadata'").addField("'updatedDate'").setOperation("<=").setVal(lastUpdateDate.toString());
      criterion.addCriterion(c1, "AND", c2);
      pgClientFactory.getInstance(tenantId).get(TABLE, FileDefinition.class,criterion,false, promise);
```